test/check-machines-hostdevs

```diff
@@ -229,6 +229,15 @@ class TestMachinesHostDevs(machineslib.VirtualMachinesCase):
                 b.wait_not_present("#delete-resource-modal")
                 b.wait_not_present(f"#vm-subVmTest1-hostdev-{self.vm_dev_id}-product")
 
+        # Add USB devices when the VM is running
+        b.wait_in_text(f"#vm-subVmTest1-{connectionName}-state", "Running")
+        output = self.run_admin(f"virsh -c qemu:///{connectionName} nodedev-list --cap usb_device", connectionName)
```

Comment:
Thanks! This new test indeed reproduces the error on the code from main.

However, I wonder if we really need the `if` here.. Don't all our test VMs have some USB device? That's a function of the hypervisor, i.e. we need to test this on both our CI and Testing Farm. It's brittle in the sense that it could silently fail here and then the test would never run.

So I'd like to test this without the condition. I'll push an extra test commit to see what happens. Somehow the previous test run got completely stuck anyway.

The fix here [breaks testHostDevAddSessionConnection](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1546-c966aabd-20240410-085437-fedora-39-firefox/log.html#85) -- I think the test is wrong, and the bug fixed here previously just hid it. I'll take a look.

Reply:
Ah, I see this check already existed. I'll clean it up a bit to avoid duplicating it.

Reply:
I met a scenario that some VMs from ppc64le and s390x will not have USB or PCI devices at all, so that I add the if there. Also, I think assertNotEqual is also a good way, thanks.
Also, I think some virt-host-validate related fixes was merged in main by https://github.com/cockpit-project/cockpit-machines/pull/1543. Are there some duplications?
And about failures of testHostDevAddSessionConnection, I saw the screenshot, and it said permission denied. Is there a reason that the non-root user should not be allowed to attach host devices? Not sure about this.

Reply:
Alright, I tried to make it unconditional, and it indeed [does fail on the TF](https://artifacts.dev.testing-farm.io/fbda1632-72af-4006-8ae7-baba5b9af107/). Alright, I've tried :grin: I'll revert the second patch.

> Is there a reason that the non-root user should not be allowed to attach host devices?

It would be dangerous if it *did* work -- if you can't access them as user on the host, it surely can't work if you forrward it to a VM of yours. So that test case is a bit too optimistic. As user we really can only check for the "not allowed" case. I'll rearrange this a bit.

Reply:
I cleaned up the tests, they were quite messy. Both work now.